File: frontend/src/api/adminDashboard.js
import apiClient from "./client";

export async function getAdminAnalytics() {
  const response = await apiClient.get("/admin-dashboard/analytics");
  return response.data;
}

export async function generateAdminReport() {
  const response = await apiClient.post("/admin-dashboard/report");
  return response.data;
}

File: backend/app/routers/admin_dashboard.py
from fastapi import APIRouter, Depends

from app.models.admin_dashboard import AdminAnalyticsResponse, AdminReportResponse
from app.services.admin_dashboard import analytics, generate_admin_report
from app.utils.dependencies import require_roles

router = APIRouter(prefix="/admin-dashboard", tags=["Admin Dashboard"])


@router.get("/analytics", response_model=AdminAnalyticsResponse)
async def get_analytics(_: object = Depends(require_roles("admin"))):
    return await analytics()


@router.post("/report", response_model=AdminReportResponse)
async def create_report(_: object = Depends(require_roles("admin"))):
    return await generate_admin_report()
